source/extensions/stat_sinks/hystrix/hystrix.cc

```diff
@@ -297,13 +313,34 @@ Http::Code HystrixSink::handlerHystrixEventStream(absl::string_view,
   return Http::Code::OK;
 }
 
-void HystrixSink::flush(Stats::Source&) {
+void HystrixSink::flush(Stats::Source& source) {
   if (callbacks_list_.empty()) {
     return;
   }
   incCounter();
   std::stringstream ss;
   Upstream::ClusterManager::ClusterInfoMap clusters = server_.clusterManager().clusters();
+
+  // Save a map of the relevant histograms per cluster in a convenient format.
+  std::unordered_map<std::string, QuantileLatencyMap> time_histograms;
+  for (const Stats::ParentHistogramSharedPtr histogram : source.cachedHistograms()) {
+    // histogram->name() on clusters of the format "cluster.cluster_name.histogram_name"
+    // i.e. "cluster.service1.upstream_rq_time".
+    const std::vector<std::string> split_name = absl::StrSplit(histogram->name(), '.');
```

Comment:
This can still be a vector of `string_view`s.  If we need to put them into a map, it's easy to convert them when needed.  This means we only need to store them as strings if we need them to be permanent.

Reply:
took your suggestion to use tags, removed this vector.